org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/codemining/DomainmodelCodeMiningProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * generated by Xtext
+ */
+package org.eclipse.xtext.example.domainmodel.ui.codemining;
+
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.codemining.ICodeMining;
+import org.eclipse.xtext.EcoreUtil2;
+import org.eclipse.xtext.Keyword;
+import org.eclipse.xtext.common.types.JvmOperation;
+import org.eclipse.xtext.example.domainmodel.domainmodel.Operation;
+import org.eclipse.xtext.example.domainmodel.services.DomainmodelGrammarAccess;
+import org.eclipse.xtext.nodemodel.ICompositeNode;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.ui.codemining.AbstractXtextCodeMiningProvider;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.util.IAcceptor;
+import org.eclipse.xtext.xbase.jvmmodel.IJvmModelAssociations;
+
+import com.google.inject.Inject;
+
+public class DomainmodelCodeMiningProvider extends AbstractXtextCodeMiningProvider {
+	@Inject
+	private IJvmModelAssociations jvmModelAssociations;
+	@Inject
+	private DomainmodelGrammarAccess grammar;
+	
+	@Override
+	protected void createLineHeaderCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator, IAcceptor<ICodeMining> acceptor) 
+			throws BadLocationException{
+		int lineCount = document.getNumberOfLines();
+		for (int i = 0; i < lineCount; i++) {
+			String line = document.get(document.getLineOffset(i), document.getLineLength(i)).trim();
+			if(line.contains("op")) {
```

Comment:
We should fix the example

Reply:
in which regard?

Reply:
line.contains("op") is a rather weak constraint, e.g. Properties names 'propertyName' or 'top' won't work.